pkg/volume/glusterfs/glusterfs.go

```diff
@@ -797,7 +797,7 @@ func (p *glusterfsVolumeProvisioner) createEndpointService(namespace string, epS
 			Namespace: namespace,
 			Name:      epServiceName,
 			Labels: map[string]string{
-				"gluster.kubernetes.io/provisioned-for-pvc": pvcname,
+				"gluster.kubernetes.io/provisioned-for-pvc": pvcID,
```

Comment:
Please use different label for UID, `gluster.kubernetes.io/provisioned-for-pvc-uid`?
And add the PVC name to annotation, so user can see what created this endpoint/service.